src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/StudentQueryHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.StudentDto;
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.StudentIdentification;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@Component
+@FieldDefaults(makeFinal = true)
+@RequiredArgsConstructor
+class StudentQueryHandler {
+
+    StudentQueryRepository studentQueryRepository;
+    StudentMapper studentMapper;
+
+    List<StudentIdentification> findAllStudents(int page, int size) {
+        return studentQueryRepository.findAll(getPageRequest(page, size))
+                .getContent().stream()
+                .map(studentMapper::toStudentIdentification)
+                .toList();
+    }
+
+    private PageRequest getPageRequest(int page, int size) {
+        return PageRequest.of(page, size, Sort.by(
+                Sort.Order.asc("lastName"),
+                Sort.Order.asc("firstName"),
+                Sort.Order.asc("email")
+        ));
+    }
+
+    Optional<StudentDto> findStudentById(UUID studentId) {
```

Comment:
Można byłoby jeszcze rozdzielić view model od modelu biorącego udział w modyfikacjach, ale w przypadku tego projektu byłby to overkill, także to bardziej informacyjnie.